faucet.go

```diff
@@ -818,3 +875,94 @@ func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
 		log.Errorf("unable to render home page: %v", err)
 	}
 }
+
+// payInvoice is a hybrid http.Handler that handles: the validation of the
+// pay invoice form, rendering errors to the form, and finally streaming the
+// payment if all the parameters check out.
+func (l *lightningFaucet) payInvoice(homeTemplate *template.Template,
+	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
+
+	elapsed := time.Since(lastPayInvoiceTime)
+	if elapsed < PayInvoiceTimeout {
+		homeState.SubmissionError = PayInvoiceTimeNotElapsed
+		homeTemplate.Execute(w, homeState)
+		return
+	}
+	lastPayInvoiceTime = time.Now()
+
+	// Get the invoice and try to verify and decode.
+	ctxb := context.Background()
+	payReq := r.FormValue("payinvoice")
+	payReqString := &lnrpc.PayReqString{PayReq: payReq}
```

Comment:
Perform a `strings.TrimSpace()` before this line, since spaces in the begining/end of the payment request are common when pasting from the clipboard.